### PR TITLE
Clean Google Dependencies

### PR DESCRIPTION
**Pull request by abhioncbr**
As per the [issue](https://github.com/apache/pinot/issues/13077). Here are the details of the changes.

- The Following properties in pom.xml are useless, and hence, removing it
   ```bash
    <google.auth.version>1.23.0</google.auth.version>
    <gson.version>2.10.1</gson.version>
    <guava.version>33.1.0-jre</guava.version>
    <google.http-client.version>1.44.1</google.http-client.version>
   ```
Indirectly, the values of the libraries mentioned above come from this [file](https://github.com/googleapis/sdk-platform-java/blob/f122c6f3beebd005a257fe687cf018baf7673b63/gapic-generator-java-pom-parent/pom.xml#L28), and we are not able to override the maven properties values.
- Deleted unnecessary `google-api-services-storage` dependency. 
- moved a couple of Google libraries hardcoded versions to property-based.

**NOTE**
- Also, I would like to highlight that we are compiling proto based on the following two property values
  ```bash
   <protobuf.version>3.25.2</protobuf.version>
   <grpc.version>1.61.1</grpc.version> 
   ```
However, runtime `grpc` and `protobuf` versions come from the same file linked above, i.e. (`1.62.2` and `3.25.3`). In future, if we want to update the Google storage dependency version, we need to ensure that the compiled proto version has no issues with runtime grpc and protobuf libraries.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1395,7 +1389,7 @@
       <dependency>
         <groupId>net.openhft</groupId>
         <artifactId>posix</artifactId>
-        <version>2.26ea0</version>
+        <version>2.25ea0</version>
```
Comment (Jackie-Jiang) [inline pom.xml:1392]: Is this change intended?

File: pom.xml [hunk 2]
```diff
@@ -1646,7 +1640,7 @@
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-javadoc-plugin</artifactId>
-          <version>3.7.0</version>
+          <version>3.6.3</version>
```
Comment (Jackie-Jiang) [inline pom.xml:1643]: Is this change intended?

File: pom.xml [hunk 3]
```diff
@@ -224,12 +224,12 @@
...
-    <google.auth.version>1.23.0</google.auth.version>
-    <gson.version>2.10.1</gson.version>
-    <guava.version>33.1.0-jre</guava.version>
-    <google.http-client.version>1.44.1</google.http-client.version>
-    <protobuf.version>3.25.2</protobuf.version>
-    <grpc.version>1.61.1</grpc.version>
+    <protobuf.version>3.25.2</protobuf.version> <!-- only getting used in proto compilation plugin -->
+    <grpc.version>1.61.1</grpc.version>  <!-- only getting used in proto compilation plugin -->
+    <google.findbugs.version>3.0.2</google.findbugs.version>
+    <google.errorprone.version>2.27.1</google.errorprone.version>
```
Comment (Jackie-Jiang) [inline pom.xml:230]: This doesn't match the current version

File: pom.xml [hunk 4]
```diff
@@ -224,12 +224,12 @@
...
     <!-- Google Libraries -->
     <google.cloud.libraries.version>26.40.0</google.cloud.libraries.version>
-    <google.auth.version>1.23.0</google.auth.version>
-    <gson.version>2.10.1</gson.version>
-    <guava.version>33.1.0-jre</guava.version>
-    <google.http-client.version>1.44.1</google.http-client.version>
-    <protobuf.version>3.25.2</protobuf.version>
-    <grpc.version>1.61.1</grpc.version>
+    <protobuf.version>3.25.2</protobuf.version> <!-- only getting used in proto compilation plugin -->
+    <grpc.version>1.61.1</grpc.version>  <!-- only getting used in proto compilation plugin -->
```
Comment (Jackie-Jiang) [inline pom.xml:228]: Is there a way to extract these 2 versions from the BOM?

Comment (abhioncbr) [reply pom.xml]: Imported `grpc-bom` and `protobuf-pom` into the project. 

In this way, the version during compilation and runtime should be the same. However, while updating `google-bom,` we need to make sure that the `grpc` and `protobuf ` versions is supported or not.

**Review by Jackie-Jiang — APPROVED** [review]
